paimon-core/src/main/java/org/apache/paimon/stats/Stats.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.stats;
+
+import org.apache.paimon.fs.FileIO;
+import org.apache.paimon.fs.Path;
+import org.apache.paimon.schema.TableSchema;
+import org.apache.paimon.types.DataType;
+import org.apache.paimon.utils.JsonSerdeUtil;
+import org.apache.paimon.utils.OptionalUtils;
+
+import org.apache.paimon.shade.jackson2.com.fasterxml.jackson.annotation.JsonCreator;
+import org.apache.paimon.shade.jackson2.com.fasterxml.jackson.annotation.JsonInclude;
+import org.apache.paimon.shade.jackson2.com.fasterxml.jackson.annotation.JsonProperty;
+
+import javax.annotation.Nullable;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+import java.util.OptionalLong;
+
+/**
+ * Global stats, supports the following stats.
+ *
+ * <ul>
+ *   <li>mergedRecordCount: the total number of records after merge
+ *   <li>mergedRecordSize: the size of the mergedRecordCount in bytes
+ *   <li>colStats: column stats map
+ * </ul>
+ */
+public class Stats {
+
+    // SnapshotId that stats belongs to
+    private static final String FIELD_SNAPSHOT_ID = "snapshotId";
+    private static final String FIELD_MERGED_RECORD_COUNT = "mergedRecordCount";
+    private static final String FIELD_MERGED_RECORD_SIZE = "mergedRecordSize";
+    private static final String FIELD_COL_STATS = "colStats";
+
+    @JsonProperty(FIELD_SNAPSHOT_ID)
+    private final long snapshotId;
+
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    @JsonProperty(FIELD_MERGED_RECORD_COUNT)
+    private final @Nullable Long mergedRecordCount;
+
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    @JsonProperty(FIELD_MERGED_RECORD_SIZE)
+    private final @Nullable Long mergedRecordSize;
+
+    @JsonProperty(FIELD_COL_STATS)
+    private final Map<String, ColStats<?>> colStats;
```

Comment:
We need to take `schema evolution` into consideration. So the `key` of this map should contain the id of column. Therefore, the version of schema which is used in this statistics should be kept.